packages/fuels-core/src/codec/function_selector.rs

```diff
@@ -13,23 +16,35 @@ pub fn resolve_fn_selector(name: &str, inputs: &[ParamType]) -> ByteArray {
     first_four_bytes_of_sha256_hash(&fn_signature)
 }
 
+#[cfg(feature = "experimental")]
+pub fn resolve_fn_selector(name: &str, _inputs: &[ParamType]) -> Vec<u8> {
+    let bytes = name.as_bytes().to_vec();
+    let len = bytes.len() as u64;
+
+    [len.to_be_bytes().to_vec(), bytes].concat()
+}
```

Comment:
I'm out of the loop here. Why are we changing the way we resolve the fn selector? Also, why are we not using `inputs`? Shouldn't we remove it, then?

Reply:
The way how we call contracts changed with the new encoding. We do not encode the fn selector as before but use the method name instead. I left the inputs for now to avoid adding cfg flags all over our abigen code generation. Once we remove the old fn-selector behavior we can refactor this.

Reply:
I created this issue to keep track of the refactoring changes : https://github.com/FuelLabs/fuels-rs/issues/1318